Domaca01/scripts/demo.jl

```diff
@@ -0,0 +1,226 @@
+#' # DN1: SOR iteracija za razpršene matrike
+#' Avtor: Patrik Schembri
+
+#' V nalogi se ukvarjamo s SOR iteracijo na razpršenih matrikah. 
+#' Matrika hrani svoje vrednosti v matrikah `V` in `I`, ki sta velikosti n x m. Pri tem velja:
+#' $$
+#' V(i,j) = A(i, I(i, j))
+#' $$
+
+using Domaca01, LinearAlgebra, Plots
+
+#' Definiramo nov tip - Razpršeno matriko A.
+V = [[1, 2], [3, 4, 5]]
+I = [[1, 2], [1, 2, 3]]
+A = RazprsenaMatrika(V, I)
+
+#' Ker gre za poseben tip matrike moramo uvesti nove funkcije za naslenje funkcionalnosti:
+
+#' Funkcija getindex
+ix = getIndex(A, 1, 2)
+
+#' Funkcija setindex
+setindex!(A, 5, 1, 2)
+A
+
+#' Fukcija firstindex
+first = firstindex(A, 1)
+
+#' Funkcija lastindex
+last = lastIndex(A, 2)
+
+#' ## SOR iteracija
+#' Sedaj lahko izvedemo Successive over-relaxation oz. SOR iteracijo. SOR iteracija se uporablja predvsem za
+#' reševanje sistemov `Ax = b`. Pri tem upošteva tudi relaksacijski vektor ω, ki je po navadi v mejah med
+#' 0 in 2. Iteracija deluje po naslednji formuli:
+#' 
+#' $$
+#' x_i^{k+1} = (1 - omega) * x_i^{k} + omega / A_{(i, i)} * (b_i - sum_{(j<i)} A_{(ij)} * x_j^{k+1} - sum_{(j>i)} A_{(ij)} * x_j^{k})
+#' $$
+#' 
+#' Iteracija se konča ko dosežemo konvergenco po naslednjem kriteriju, kjer za toleranco vzamemo
+#' zelo majhno vrednost (npr. 1e-10):
+#' ``` |Ax(k) - b| ∞ < toleranca ```
```

Comment:
Mogoče raje uporabite LaTeX zapis $\Vert A x_k - b \Vert_\infty < tol$
```suggestion
#' $$ \Vert Ax_k - b\Vert_\infty < tol $$
```